SAMtags.tex

```diff
@@ -251,20 +255,20 @@ \subsection{Barcodes}
 Same encoding as {\sf QUAL}, i.e., Phred score + 33.
 In the case of multiple unique molecular identifiers (e.g., one on each end of the template) the recommended implementation concatenates all the quality strings with spaces (`{\tt \textvisiblespace}') between the different strings from the same template. 
 
-\item[RX:Z:\tagvalue{sequence+}]
-Sequence bases from the unique molecular identifier. 
-These could be either corrected or uncorrected. Unlike {\tt MI}, the value may be non-unique in the file. 
-Should be comprised of a sequence of bases. 
-In the case of multiple unique molecular identifiers (e.g., one on each end of the template) the recommended implementation concatenates all the barcodes with a hyphen (`{\tt -}') between the different barcodes.
+\item[CB:Z:\tagvalue{str}]
+Cell identifier based on the cellular-barcode sequence ({\tt CR}), with a correction optionally applied. The sequence may optionally be followed by a dash (`{\tt -}') and one or more alphanumeric characters to form an identifier.
+All reads with the same {\tt CB} tag putatively derive from the same cell.
 
-If the bases represent corrected bases, the original sequence can be stored in {\tt OX} (similar to {\tt OQ} storing the original qualities of bases.)
+\item[CR:Z:\tagvalue{str}]
+Cellular barcode. Sequence bases of the cellular-barcode as reported by the sequencer (uncorrected) with the respective quality scores stored in {\tt CY}.
+All reads with the same {\tt CR} tag putatively derive from the same cell.
```

Comment:
Hi! Thank you for the feedback. Can't the same observation about the sequencing errors be made about other barcodes?

Reply:
My interpretation of @yfarjoun's comments is that it isn't a 1:1 relationship, rather than it being due to errors.  Ie the same barcode can be used in multiple cells, but the reverse is not true (in each cell only one barcode will be used).

I'm not processing 10X data myself so I can't verify this, but having spoken with those that are right now the number of barcodes is typically much higher than the number of cells being used, but a barcode being shared is still possible (and likely given the birthday paradox).

Reply:
@jkbonfield  - correct. In the latest single-cell RNA-seq setup there are 737K possible barcodes, which are sample 'with replacement' into 100K droplets. Then a user-defined number of cells are populated into those droplets (we recommend < 10K cells). So you can have two cells sharing the same barcode because:
1) they ended up in the same droplet by chance
2) they ended up in different droplets, but the barcode in those droplets was the same by chance.
So the user really needs to understand those parameters to compute the probability that the CR tag is referencing a unique cell. Some platforms also have the ability to image the droplets and filter out barcodes that contain > 1 cell.  Various other schemes are being worked to identify doublets.  Here's a slightly more verbose proposal:
`All reads with the same {\tt CR} tag a likely to derive from the same cell. The probability that a given {\tt CR} value contains reads from multiple cells may could be substantial (>10%) and is technology and experiment dependent. All reads from any given cell are likely to have the same {\tt CR} tag, but not due to sequencing errors or technical artifacts in library preparation`

Reply:
Thanks @pmarks for looking over this PR and commenting.

I think the new description does rather muddy things a bit regarding more than one barcode for the same cell.  I understand the reporting of this barcode (ie sequencing) may cause differences, but it's still the same original barcode.   I'm not sure what the "technical artifacts" would be?  Are we just talking about things going wrong or mistakes?  If so it doesn't really belong in the spec description.

Reply:
@jkbonfield most protocols have some level of 'barcode exchange' that causes the cell-barcodes to 'hop' between library molecules (analogous to index-hopping on patterned flowcells). In a simple model, each read comes from the 'correct' cell with probability 1-e, and comes randomly from the whole pool of molecules with probability e.  This is what I was thinking about when I wrote 'technical artifacts'.  That's probably just a generic consideration that doesn't belong in the spec.